### PR TITLE
Check type of terminator before attempting to encode

### PR DESCRIPTION
**Pull request by JasperCraeghs**
When using `Read Until  | terminator=127`  using robotframework==3.2.1, `terminator` is a bytes object instead of a string. No encoding is needed in this case. The type check I added avoids `AttributeError: 'bytes' object has no attribute 'encode'`.

### CODE REVIEW

**Comment by Dragonwilds** [comment]
Why has this pull request not been merged yet?

**Comment by JasperCraeghs** [comment]
We're waiting on @whosaysni  to merge this PR but I'm afraid he's become inactive.

**Comment by Dragonwilds** [comment]
> 
> 
> We're waiting on @whosaysni to merge this PR but I'm afraid he's become inactive.

@JasperCraeghs Any idea why I'm receiving "TypeError: read_until() got an unexpected keyword argument 'terminator'" with your branch?

**Comment by JasperCraeghs** [comment]
> @JasperCraeghs Any idea why I'm receiving "TypeError: read_until() got an unexpected keyword argument 'terminator'" with your branch?

No, I don't. The full traceback might help.

EDIT: I might have an idea. Now that you're using my branch you now reach [L530](https://github.com/whosaysni/robotframework-seriallibrary/pull/12/files#diff-79264fbc52746534316cbea04d8f02a2cebc0d47ff089bea2a5d17393968a164R530). `read_until` of a lower-layer library is now called, but the problem is that this library's function doesn't have a keyword argument `terminator`. In case of pyserial, the keyword argument is called `expected`. See https://github.com/pyserial/pyserial/blob/master/serial/serialutil.py#L654.
This renaming was done in 2018 in [this commit](https://github.com/pyserial/pyserial/commit/daaf33edd1a054a52fa8ec297a5767d06f96a942) and [this PR](https://github.com/pyserial/pyserial/pull/356).

TL;DR: Downgrade pyserial to [`pyserial==3.4`](https://pypi.org/project/pyserial/3.4/), try again and open an issue or implement a fix if that fixed it.

**Comment by lesnake** [comment]
@JasperCraeghs  I'm trying your commits, but they don't pass unit tests provided.

The failing test case is `Read Until should read until terminator or size`, when using read_until with CD as terminator.
```
Read Until should read until terminator or size                       | FAIL |
AB CD EF != AB CD
```
`robot  tests/acceptance/test_library.robot`

Can you check on your side ?

**Comment by whosaysni** [comment]
Thanks, 0.4.1+ fixed this issue.
